code_snippets/kotlin/src/main/kotlin/usecases/asynctasks/synctoasync/MyClient.kt

```diff
@@ -0,0 +1,40 @@
+package usecases.asynctasks.synctoasync
+
+import dev.restate.sdk.client.Client
+import develop.workflows.Email
+import usecases.asynctasks.synctoasync.DataPreparationServiceClient.IngressClient
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
+
+// <start_here>
+class MyClient {
+
+    companion object {
+        private val rs: Client = Client.connect("http://localhost:8080")
+    }
+
+    suspend fun downloadData(userId: String, email: Email) {
+        // <mark_1>
+        val client: IngressClient = DataPreparationServiceClient.fromClient(rs, userId)
+        // </mark_1>
+
+        // <mark_2>
+        client.submit(userId)
+        // </mark_2>
+
+        try {
+            // <mark_3>
+            CompletableFuture.anyOf(client.workflowHandle().attachAsync())
+                .orTimeout(30, TimeUnit.SECONDS)
+                .join()
```

Comment:
Use kotlin stuff for this, not `CompletableFuture`

Reply:
All the client methods have a variant `suspend`, e.g. `attachSuspend`, which you import from `dev.restate.sdk.kotlin`. For timeouts: https://kotlinlang.org/docs/cancellation-and-timeouts.html#timeout